app/src/androidTest/java/com/android/voyageur/ui/trip/schedule/WeeklyViewScreenTest.kt

```diff
@@ -0,0 +1,196 @@
+package com.android.voyageur.ui.trip.schedule
+
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import com.android.voyageur.model.activity.Activity
+import com.android.voyageur.model.activity.ActivityType
+import com.android.voyageur.model.location.Location
+import com.android.voyageur.model.trip.Trip
+import com.android.voyageur.ui.navigation.NavigationActions
+import com.android.voyageur.ui.navigation.Route
+import com.google.firebase.Timestamp
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.util.Date
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.*
+
+class WeeklyViewScreenTest {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private lateinit var navigationActions: NavigationActions
+  private lateinit var mockTrip: Trip
+  private var isDailyViewSelected = false
+
+  @Before
+  fun setUp() {
+    navigationActions = Mockito.mock(NavigationActions::class.java)
+
+    // Mock current route for navigation actions
+    `when`(navigationActions.currentRoute()).thenReturn(Route.TOP_TABS)
+
+    // Create mock trip with activities
+    mockTrip =
+        Trip(
+            id = "test-trip",
+            name = "London Trip",
+            startDate =
+                Timestamp(Date.from(LocalDateTime.of(2024, 10, 3, 0, 0).toInstant(ZoneOffset.UTC))),
+            endDate =
+                Timestamp(Date.from(LocalDateTime.of(2024, 11, 4, 0, 0).toInstant(ZoneOffset.UTC))),
+            activities =
+                listOf(
+                    Activity(
+                        "1",
+                        "",
+                        Location("", "", "", ""),
+                        startTime =
+                            Timestamp(
+                                Date.from(
+                                    LocalDateTime.of(2024, 10, 3, 14, 0)
+                                        .toInstant(ZoneOffset.UTC))),
+                        endDate =
+                            Timestamp(
```

Comment:
Here, endDate is now endTime, can you adjust the changes?

Reply:
Thank you for making me notice these changes! I fixed this in the code.